include/mra/kernels/transform.h

```diff
@@ -10,11 +10,11 @@
 #include "mra/tensor/tensorview.h"
 namespace mra {
 
-  template <Dimension NDIM, typename T>
+  template < template < typename, Dimension> class TensorView, Dimension NDIM, typename T>
   SCOPE void transform(
-    const TensorView<T, NDIM>& t,
-    const TensorView<T, 2>& c,
-    TensorView<T, NDIM>& result,
+    const TensorView<T, NDIM>&& t,
+    const TensorView<T, 2>&& c,
+    TensorView<T, NDIM>&& result, /* make arguments universal references*/
```

Comment:
Does this work with template-template parameters? I think it would be easier to have a regular template parameter TensorViewT and take that as TensorViewT&&. We can add a constraint later to check that it's actually a TensorView.